Update Configuration.md Documentation (#282)
node-red/node-red.github.io: docs/user-guide/runtime/configuration.md

- Security
- Server 
- Runtime
- Editor
- Node
 
### Flow File and User Directory

**flowFile**
: the file used to store the flows. If not set, defaults to `flows_<hostname>.json`<br /> Default: `flowFile: 'flows.json',`

**credentialSecret**
: By default, credentials are encrypted in storage using a generated key. To specify your own secret, set the following property. If you want to disable encryption of credentials, set this property to false. Note: once you set this property, do not change it - doing so will prevent node-red from being able to decrypt your existing credentials and they will be lost. Example: `credentialSecret: "a-secret-key",`

**flowFilePretty**
:  By default, the flow JSON will be formatted over multiple lines making
     it easier to compare changes when using version control.
     To disable pretty-printing of the JSON set the following property to false. Default: `flowFilePretty: true`

**userDir**
: By default, all user data is stored in a directory called `.node-red` under the user's home directory. To use a different location, the following property can be used `userDir: '/home/nol/.node-red/',` Default: `$HOME/.node-red`

> **nodesDir**
> *Since Node-RED 3.0.0.*
> : a directory to search for additional installed nodes. Node-RED searches the `nodes`
>   directory under the *userDir* directory. This property allows an additional directory
>   to be searched, so that nodes can be installed outside of the Node-RED install
>   structure. Default: `$HOME/.node-red/nodes`

Reviewer: ```suggestion
**nodesDir**
: This property allows an additional directory to be searched for nodes to import.

*Before Node-RED 3.0.0*
: This setting can be a single path `string` only

*Since Node-RED 3.0.0*
: This setting can be...
* A single path `string`
* An array of  path `string`s
```

I think I forgot to merge the updated comments for `nodesDir` when i raised the PR (BAH!) 
the above (or if you can think of better words for "it can be a string or array of strings") is more up-to-date.

Author: `nodesDir`  is a good example of some settings options that have changed with added features from version to version. If we add how it was Before and after v3 it could start getting messy (can of worms). Personally i think the docs should inform about the configuration as it stands now with the latest Node-red version. and add to the note at the top : "*Please note that some of the options may not apply for older versions of Node-red. (**consult the comments in your settings.js file**)*"

     *      then "/home/nol/docs" will be served at "/static/"
     * e.g. if httpStatic = [{path: '/home/nol/pics/', root: "/img/"}]
     *      and httpStaticRoot = "/static/"
     *      then "/home/nol/pics/" will be served at "/static/img/"
     */
    httpStaticRoot: '/static/',
```

  *Standalone only*.

**httpAdminAuth**
: *Deprecated*: see `adminAuth`.

  enables HTTP Basic Authentication on the editor UI:

```javascript
    httpAdminAuth: {user:"nol", pass:"5f4dcc3b5aa765d61d8327deb882cf99"}
```

The `pass` property is the md5 hash of the actual password. The following
 command can be used to generate the hash:

    node -e "console.log(require('crypto').createHash('md5').update('YOUR PASSWORD HERE','utf8').digest('hex'))"

  *Standalone only*.

**httpRoot**
: *Deprecated*: see `httpAdminRoot` and `httpNodeRoot`.

### Runtime

**lang**
*Since Node-RED 2.0.0.*
: the following option is to run node-red in your preferred language.<br/> 
Available languages include: `en-US (default), ja, de, zh-CN, zh-TW, ru, ko`. Some languages are more complete than others.

**diagnostics**
*Since Node-RED 3.0.0.*
: Configure diagnostics options.

```javascript

>     diagnostics: {
>         /** @type {boolean} enable or disable diagnostics. Must be set to `false` to disable */
>         enabled: true,
>         /** @type {"basic"|"admin"} diagnostic level can be "basic" (default) or "admin" (more sensitive details are included) */
>         level: "basic",
>     },

Reviewer: ```suggestion
    diagnostics: {
        /** enable or disable diagnostics endpoint. Must be set to `false` to disable */
        enabled: true,
        /** enable or disable diagnostics display in the node-red editor. Must be set to `false` to disable */
        ui: true,
    },
```

that description is out of date

```

> When `enabled` is `true` (or unset), diagnostics data will be available at http://localhost:1880/diagnostics . When `level` is "basic" (or unset), the diagnostics will not include sensitive data. Set level to "admin" for detailed diagnostics.  

Reviewer: ```suggestion
When `enabled` is `true` (or unset), diagnostics data will be available at `http://localhost:1880/diagnostics`.  
When `ui` is `true` (or unset), diagnostics data will accessible in the Node-RED Action List as "Show System Info" 
```

> **runtimeState**
> *Since Node-RED 3.0.0.*
> : enable or disable flows/state
> 
> ```javascript
>     runtimeState: {
>         /** enable or disable flows/state endpoint. Must be set to `false` to disable */
>         enabled: false,
>         /** show or hide runtime stop/start options in the node-red editor. Must be set to `false` to hide */
>         ui: false,
>     },
> ```

Reviewer: ```suggestion
**runtimeState**
*Since Node-RED 3.0.0.*
: enable or disable the ability to Start and Stop the runtime

```javascript
    runtimeState: {
        /** enable or disable this feature. */
        enabled: false,
        /** show or hide runtime stop/start options in the node-red editor. */
        ui: false,
    },
```
```

        },
        header: {
            title: "Node-RED",
            image: "/absolute/path/to/header/image", // or null to remove image
            url: "http://nodered.org" // optional url to make the header text/image a link to this url
        },
        deployButton: {
            type:"simple",
            label:"Save",
            icon: "/absolute/path/to/deploy/button/image" // or null to remove image
        },
        menu: { // Hide unwanted menu items by id. see packages/node_modules/@node-red/editor-client/src/js/red.js:loadEditor for complete list
            "menu-item-import-library": false,
            "menu-item-export-library": false,
            "menu-item-keyboard-shortcuts": false,
            "menu-item-help": {
                label: "Alternative Help Link Text",
                url: "http://example.com"
            }
        },
        tours: false, // disable the Welcome Tour for new users
        userMenu: false, // Hide the user-menu even if adminAuth is enabled
        login: {
            image: "/absolute/path/to/login/page/big/image" // a 256x256 image
        },
        logout: {
            redirect: "http://example.com"

>         },

Reviewer: This info seems to be missing from your PR?

Author: yes, im glad you noticed .. `editorTheme` is one of the sections that i wanted your input on. `editorTheme` is also one of the options that has added features, so i wasnt sure how you wanted to handle options that may not apply for older versions ..  i'll update the description if its ok to reflect the latest v3

Author: Steve some of those options seem to be missing from the latest `settings.js` file also ?
i didnt know what sub-options still apply for the latest NR version.

Reviewer: I can't answer why Andy. I know Nick did a good deal of tidy up and may have moved these or deleted them as they should be perhaps? 
@knolleary can you comment on this one please?
